src/pyhf/utils.py

```diff
@@ -109,17 +147,48 @@ def qmu(mu, data, pdf, init_pars, par_bounds):
         Float: The calculated test statistic, :math:`q_{\mu}`
     """
     tensorlib, optimizer = get_backend()
-    mubhathat = optimizer.constrained_bestfit(
-        loglambdav, mu, data, pdf, init_pars, par_bounds
-    )
-    muhatbhat = optimizer.unconstrained_bestfit(
-        loglambdav, data, pdf, init_pars, par_bounds
-    )
-    qmu = loglambdav(mubhathat, data, pdf) - loglambdav(muhatbhat, data, pdf)
+    mubhathat, muhatbhat, qmu = pll_and_estimators(mu, data, pdf, init_pars, par_bounds)
     qmu = tensorlib.where(muhatbhat[pdf.config.poi_index] > mu, [0], qmu)
     return qmu
 
 
+def q0(mu, data, pdf, init_pars, par_bounds):
```

Comment:
why does this have a `mu` parameter? shouldn't that be fixed to 0?

Reply:
mhh - probably there is no real use in having this as a free parameter. On the other hand - when we want to switch to selecting the test statistic from a mapping it might be simpler when they all have the same parameters.

Reply:
I think a way to do this is to have a kind of test statistic factory which takes kwargs that could be 

```
def get_teststat(type, **opts):
    ..
  teststat = get_teststat(type = 'qmu', mu = mu)
  teststat(data, pdf)

or 

  teststat = get_teststat(type = 'q0')
  teststat(data, pdf)

```

i.e. `mu` is not part of the test stat API (which just takes the data and the pdf, but part of the API that gives you the teststat